### PR TITLE
add_key code

### PR DESCRIPTION
**Pull request by unforced**
Made a basic add_key page.

### CODE REVIEW

File: kraller/kraller.py [hunk 1]
```diff
@@ -194,7 +196,37 @@ def signup():
 @app.route('/add_key', methods=['POST'])
 @requires_auth
 def add_key():
-    pass
+    username = session['username']
+    form = AddKeyForm()
+    if not form.validate_on_submit():
+        flash('There was an error submitting the form!')
+        return render_template('signup.tmpl', form=form)
```
Comment (imbstack) [inline kraller/kraller.py]: Remind me how this works.  Does `/signup` handle all of the GETs and such?  Is `signup.tmpl` still the template we want here?

**Comment by imbstack** [comment]
lgtm.  

Just let me know the answer to my question in there and we can merge this in.

File: kraller/kraller.py (hunk 1 above)
Comment (unforced) [inline kraller/kraller.py]: Oh, that should probably render add_key.tmpl.  Failed copy paste I forgot to change.
/signup handles the GETS and if the user account exists, it renders add_key.tmpl. add_key.tmpl then POSTS to add_key which goes to this code.

File: kraller/kraller.py [hunk 2]
```diff
@@ -194,7 +196,37 @@ def signup():
 @app.route('/add_key', methods=['POST'])
 @requires_auth
 def add_key():
-    pass
+    username = session['username']
+    form = AddKeyForm()
+    if not form.validate_on_submit():
+        flash('There was an error submitting the form!')
+        return render_template('add_key.tmpl', form=form)
```
Comment (imbstack) [inline kraller/kraller.py:212]: Alright, now what if I do this:
1. Go to `/signup` after I already have an account.
2. Put in a new key and hit submit.
3. There is an error with validation and I am kicked back out to `/add_key`.
4. I refresh the page.

Shouldn't this be a redirect to `/signup` or allow `add_key` to handle GETs?

Comment (imbstack) [inline kraller/kraller.py:212]: I guess it's fine.  I would just prefer that `/add_key` isn't an actual page that people land on, but just something that redirects away.

Or rather, it's even weirder that `/signup` renders the template for `add_key`.

Comment (unforced) [inline kraller/kraller.py:212]: Oh ya, you're probably right. Guess I need to do actual stuff then. ;(

**Comment by imbstack** [comment]
Looks good to me.  Shipping it.

**Pull request merged**